packages/tkg-clusterclass-aws/bundle/config/upstream/base.yaml

```diff
@@ -1146,6 +1158,73 @@ spec:
             content: {{or $proxy $image}}
             encoding: base64
             permissions: "0444"
+  - name: apiserverEncryptionConf
+    definitions:
+      - selector:
+          apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+          kind: KubeadmControlPlaneTemplate
+          matchResources:
+            controlPlane: true
+          jsonPatches:
+            - op: add
+              path: /spec/template/spec/kubeadmConfigSpec/files
+              valueFrom:
+                template: |
+                  - path: /etc/kubernetes/encryption-config.yaml
+                    owner: "root:root"
+                    permissions: "0600"
+                    encoding: base64
+                    content: {{ $apiserverEncryptionConf }}
+            - op: add
+              path: /spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraArgs
+              valueFrom:
+                template: |
+                  encryption-provider-config: "/etc/kubernetes/encryption-config.yaml"
+            - op: add
+              path: /spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraVolumes
+              valueFrom:
+                template: |
+                  - name: encryption-conf
+                    hostPath: /etc/kubernetes/encryption-config.yaml
+                    mountPath: /etc/kubernetes/encryption-config.yaml
+  - name: apiserverEventRateLimitConf
+    definitions:
+      - selector:
+          apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+          kind: KubeadmControlPlaneTemplate
+          matchResources:
+            controlPlane: true
+          jsonPatches:
+            - op: add
+              path: /spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraArgs
+              valueFrom:
+                template: |
+                  enable-admission-plugins: "NodeRestriction,EventRateLimit"
+                  admission-control-config-file: "/etc/kubernetes/admission-control-config.yaml"
+            - op: add
+              path: /spec/template/spec/kubeadmConfigSpec/files
+              valueFrom:
+                template: |
+                  - path: /etc/kubernetes/admission-control-config.yaml
+                    owner: "root:root"
+                    permissions: "0600"
+                    encoding: base64
+                    content: YXBpVmVyc2lvbjogYXBpc2VydmVyLmNvbmZpZy5rOHMuaW8vdjEKa2luZDogQWRtaXNzaW9uQ29uZmlndXJhdGlvbgpwbHVnaW5zOgotIG5hbWU6IEV2ZW50UmF0ZUxpbWl0CiAgcGF0aDogZXZlbnRjb25maWcueWFtbAoK
```

Comment:
Could you use a ytt encode function to generate base64 content? It will be easy to review or change in the future.